Using torch.accelerator in comm_mode_features_example.py and visualize_sharding_example.py

githubsgi opened the pull request:
Continuation of https://github.com/pytorch/pytorch/pull/153213  .

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @EikanWang


 @guangyey
 @kwen2501

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

githubsgi committed:
Using torc.accelerator

guangyey commented on torch/distributed/tensor/examples/visualize_sharding_example.py:
```diff
@@ -17,6 +17,15 @@
 rank = int(os.environ["RANK"])
 
+def get_device_type() -> str:
+    return (
+        torch.accelerator.current_accelerator().type
+        if  torch.accelerator.current_accelerator() and torch.accelerator.device_count()
+        else "cpu"
+    )
```
```suggestion
    return acc.type if (acc := torch.accelerator.current_accelerator(True)) else "cpu"

```

guangyey commented on torch/distributed/tensor/examples/comm_mode_features_example.py:
```diff
@@ -28,12 +28,13 @@
-        "cuda"
-        if torch.cuda.is_available() and torch.cuda.device_count() >= 4
+        torch.accelerator.current_accelerator().type
+        if  torch.accelerator.current_accelerator() and torch.accelerator.device_count() >= 4
```
```suggestion
        if torch.accelerator.device_count() >= 4
```

guangyey committed:
Update torch/distributed/tensor/examples/comm_mode_features_example.py

guangyey committed:
Update torch/distributed/tensor/examples/visualize_sharding_example.py

guangyey commented on torch/distributed/tensor/examples/comm_mode_features_example.py:
```diff
@@ -28,12 +28,13 @@
     )
 
 
+
```
```suggestion
```

guangyey committed:
Update torch/distributed/tensor/examples/comm_mode_features_example.py

githubsgi commented on torch/distributed/tensor/examples/comm_mode_features_example.py:
This would fail as follows. 

```
  Error (MYPY) [union-attr]
    Item "None" of "device | None" has no attribute "type"

         28  |
         29  |def get_device_type() -> str:
         30  |    return (
    >>>  31  |        torch.accelerator.current_accelerator().type
         32  |        if torch.accelerator.device_count() >= 4
         33  |        else "cpu"
         34  |    )

+ echo ''
```

EikanWang replied:
`current_accelerator` could return `None`.

EikanWang replied:
The code needs to be polished a little bit.

guangyey replied:
If `torch.accelerator.device_count() >= 4` is True, `torch.accelerator.current_accelerator()` must not return `None`
I think it is OK to refine it or ignore the linter `# type: ignore[union-attr]`

githubsgi replied:
@guangyey , I think it is better  not to change the original  logic.

guangyey replied:
Sounds good.

githubsgi committed:
Rervsing some of the changes.

githubsgi committed:
Trying to placate lintrunner.

>>> Lint for torch/distributed/tensor/examples/comm_mode_features_example.py:

  Error (MYPY) [union-attr]
    Item "None" of "device | None" has no attribute "type"

         28  |
         29  |def get_device_type() -> str:
         30  |    return (
    >>>  31  |        torch.accelerator.current_accelerator().type
         32  |        if  torch.accelerator.current_accelerator() and torch.accelerator.device_count() >= 4
         33  |        else "cpu"
         34  |    )

>>> Lint for torch/distributed/tensor/examples/visualize_sharding_example.py:

  Error (MYPY) [union-attr]
    Item "None" of "device | None" has no attribute "type"

         20  |
         21  |def get_device_type() -> str:
         22  |    return (
    >>>  23  |        torch.accelerator.current_accelerator().type
         24  |        if  torch.accelerator.current_accelerator() and torch.accelerator.device_count() >= 4
         25  |        else "cpu"
         26  |    )

+ echo ''

+ echo -e '\e[1m\e[36mYou can reproduce these results locally by using `lintrunner -m origin/main`. (If you don'\''t get the same results, run \'\''lintrunner init\'\'' to update your local linter)\e[0m'
You can reproduce these results locally by using `lintrunner -m origin/main`. (If you don't get the same results, run \'lintrunner init\' to update your local linter)
+ echo -e '\e[1m\e[36mSee https://github.com/pytorch/pytorch/wiki/lintrunner for setup instructions. To apply suggested patches automatically, use the -a flag. Before pushing another commit,\e[0m'
See https://github.com/pytorch/pytorch/wiki/lintrunner for setup instructions. To apply suggested patches automatically, use the -a flag. Before pushing another commit,
+ echo -e '\e[1m\e[36mplease verify locally and ensure everything passes.\e[0m'
please verify locally and ensure everything passes.
+ RC=1
+ find torch -name '*.pyi' -exec git restore --staged -- '{}' +
+ jq --raw-output '"::\(if .severity == "advice" or .severity == "disabled" then "warning" else .severity end) file=\(.path),line=\(.line),col=\(.char),title=\(.code) \(.name)::" + (.description | gsub("\\n"; "%0A"))' lint.json
Error: Item "None" of "device | None" has no attribute "type"
Error: Item "None" of "device | None" has no attribute "type"

githubsgi committed:
Local lintrunner passed.

githubsgi committed:
Adding the device_count() >= 4 check back.

githubsgi committed:
Fixes for mpre lintrunner nitpick.

Not sure why the lintruner is concatenating  "if" and "torch" below .

    28  29 | def get_device_type() -> str:
    29  30 |     device_type = "cpu"
    29     |-    iftorch.accelerator.device_count() >= 4:
    30     |-        device_type = getattr (torch.accelerator.current_accelerator(), "type", "cpu")
        31 |+    iftorch.accelerator.device_count() >= 4:
        32 |+        device_type = getattr(torch.accelerator.current_accelerator(), "type", "cpu")
    32  33 |     return device_type

githubsgi commented:
The following errors  looks unrelated to this PR. 

[pull / linux-jammy-py3.9-clang12 / test (einops, 1, 1, linux.2xlarge) (pull_request)](https://github.com/pytorch/pytorch/actions/runs/16154449121/job/45605114110?pr=157317)
pull / linux-jammy-py3.9-clang12 / test (einops, 1, 1, linux.2xlarge) (pull_request)Failing after 219m
[pull / linux-jammy-py3.13-clang12 / test (einops, 1, 1, linux.2xlarge) (pull_request)](https://github.com/pytorch/pytorch/actions/runs/16154449121/job/45604937027?pr=157317)
pull / linux-jammy-py3.13-clang12 / test (einops, 1, 1, linux.2xlarge) (pull_request)Failing after 577m

d4l3k commented:
@pytorchbot rebase

d4l3k reviewed (approved):
LGTM

d4l3k commented:
@pytorchbot merge
